kernel/kernel-default/src/main/java/io/delta/kernel/parquet/RowConverter.java

```diff
@@ -96,17 +108,46 @@ public ColumnarBatch getDataAsColumnarBatch(int batchSize)
         return batch;
     }
 
+    /**
+     * @return true if all members were null
+     */
+    private boolean moveConvertersToNextRow(Optional<Long> fileRowIndex) {
+        long memberNullCount = Arrays.stream(converters)
+                .map(converter -> (ParquetConverters.BaseConverter) converter)
+                .map(converter -> {
+                    if (fileRowIndex.isPresent() &&
+                            converter instanceof ParquetConverters.FileRowIndexColumnConverter) {
+                        return ((ParquetConverters.FileRowIndexColumnConverter) converter)
+                                .moveToNextRow(fileRowIndex.get());
+                    } else {
+                        return converter.moveToNextRow();
+                    }
+                })
+                .filter(result -> result)
```

Comment:
?

Reply:
This is the same as it was before. It keeps all the isNull elements so we can get the nullCount.